x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/repositories/ClearCcrRestoreSessionRequest.java

```diff
@@ -40,14 +51,35 @@ public ActionRequestValidationException validate() {
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeString(sessionUUID);
+        if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
+            shardId.writeTo(out);
+        }
     }
 
     String getSessionUUID() {
         return sessionUUID;
     }
 
+    ShardId getShardId() {
+        return shardId;
+    }
+
     @Override
     public DiscoveryNode getPreferredTargetNode() {
         return node;
     }
+
+    @Override
+    public String[] indices() {
+        if (shardId == null) {
+            return null;
+        } else {
+            return new String[] { shardId.getIndexName() };
+        }
+    }
```

Comment:
Using just the index name from the shard works because CCR does _not_ support aliases. Otherwise it could be a lot of trouble to keep the original indices.